secure/storage/vault/src/lib.rs

```diff
@@ -78,6 +78,7 @@ impl From<serde_json::Error> for Error {
 /// token, but policies can be amended afterward. So you cannot add new policies to a token, but
 /// you can increase the tokens abilities by modifying the underlying policies.
 pub struct Client {
+    agent: ureq::Agent,
```

Comment:
lol

Reply:
So it looks like this was the only issue? The fact that we were having to open a new connection for each request?